src/providers/auto-complete/contacts-service.ts

```diff
@@ -5,43 +5,63 @@ import lodash from 'lodash';
 
 import { UserDataProvider } from '@providers/user-data/user-data';
 import { PublicKey } from 'ark-ts/core';
+import {AutoCompleteContact} from "@models/contact";
 
 @Injectable()
 export class ContactsAutoCompleteService implements AutoCompleteService {
-  labelAttribute = "name";
-  formValueAttribute = "address"
 
-  constructor(private userDataProvider: UserDataProvider) { }
+  // even though this fields are unused, they are required by the AutoCompleteService!
+  public labelAttribute = 'name';
+  public formValueAttribute = 'address';
 
-  getResults(keyword: string) {
+  public constructor(private userDataProvider: UserDataProvider) {
+  }
+
+  getResults(keyword: string): AutoCompleteContact[] {
     keyword = keyword.toLowerCase();
 
-    let contacts = lodash.map(this.userDataProvider.currentProfile.contacts, (value, key) => {
+    const contacts: AutoCompleteContact[] = lodash.map(this.userDataProvider.currentProfile.contacts, (value, key) => {
       if (value['name']) {
-        return { address: key.toString(), name: 'Contact: ' + value['name'].toString() };
+        return {
+          address: key.toString(),
+          name: value['name'].toString(),
+          iconName: "ios-contacts-outline"
+        } as AutoCompleteContact;
       }
     });
-    let wallets = lodash.map(this.userDataProvider.currentProfile.wallets, (value, key) => {
-      let address = value['address'];
-      let label = value['label'] || value['address'];
-      if (address && label) {
-        return { address: address.toString(), name: label.toString() };
+
+    const wallets: AutoCompleteContact[] = lodash.map(this.userDataProvider.currentProfile.wallets, (value) => {
+      const address = value['address'];
+      const label = value['label'] || value['address'];
+      if (address) {
+        return {
+          address: address.toString(),
+          name: label.toString(),
+          iconName: "ios-cash-outline"
+        } as AutoCompleteContact;
```

Comment:
`as ...` is new to me. please explain!

Reply:
It's basically a `cast` - the same as `<AutoCompleteContact>something` . However because of some problems with `tsx` files `as` is the new preferred syntax I believe. (https://stackoverflow.com/a/33503842/1712056)

=> In this case it helps to clarify what the type of my newly created anonymous object is (because it cannot be easily interferred from the method). Therefore if we would for example rename `iconName` to another variable name, it would not compile anymore, since the object wouldn't match the type anymore.

Reply:
That's nice, I didn't realise it was a thing. I'll be using that from now on. Thanks!